RecoMTD/DetLayers/src/MTDDetSector.cc

```diff
@@ -0,0 +1,177 @@
+                                                              const Propagator& prop,
+                                                              const MeasurementEstimator& est) const {
+  TrajectoryStateOnSurface ms = prop.propagate(ts, specificSurface());
+
+#ifdef EDM_ML_DEBUG
+  LogTrace("MTDDetLayers") << "MTDDetSector::compatible, sector: \n"
+                           << (*this) << "\n  TS at Z,R,phi: " << std::fixed << std::setw(14) << ts.globalPosition().z()
+                           << " , " << std::setw(14) << ts.globalPosition().perp() << " , " << std::setw(14)
+                           << ts.globalPosition().phi();
+  if (ms.isValid()) {
+    LogTrace("MTDDetLayers") << " DEST at Z,R,phi: " << std::fixed << std::setw(14) << ms.globalPosition().z() << " , "
+                             << std::setw(14) << ms.globalPosition().perp() << " , " << std::setw(14)
+                             << ms.globalPosition().phi() << " local Z: " << std::setw(14) << ms.localPosition().z();
+  } else {
+    LogTrace("MTDDetLayers") << " DEST: not valid";
+  }
+#endif
+
+  return make_pair(ms.isValid() and est.estimate(ms, specificSurface()) != 0, ms);
+}
+
+vector<GeometricSearchDet::DetWithState> MTDDetSector::compatibleDets(const TrajectoryStateOnSurface& startingState,
+                                                                      const Propagator& prop,
+                                                                      const MeasurementEstimator& est) const {
+  LogTrace("MTDDetLayers") << "MTDDetSector::compatibleDets, sector: \n"
+                           << (*this) << "\n  TS at Z,R,phi: " << std::fixed << std::setw(14)
+                           << startingState.globalPosition().z() << " , " << std::setw(14)
+                           << startingState.globalPosition().perp() << " , " << std::setw(14)
+                           << startingState.globalPosition().phi();
+
+  vector<DetWithState> result;
+
+  // Propagate and check that the result is within bounds
+  pair<bool, TrajectoryStateOnSurface> compat = compatible(startingState, prop, est);
+  if (!compat.first) {
+    LogTrace("MTDDetLayers") << "    MTDDetSector::compatibleDets: not compatible"
+                             << "    (should not have been selected!)";
+    return result;
+  }
+
+  TrajectoryStateOnSurface& tsos = compat.second;
+  GlobalPoint startPos = tsos.globalPosition();
+
+  LogTrace("MTDDetLayers") << "Starting position: " << startPos;
+
+  // determine distance of det center from extrapolation on the surface, sort dets accordingly
+
+  size_t idetMin = basicComponents().size();
+  double dist2Min = std::numeric_limits<double>::max();
+  std::vector<std::pair<double, size_t> > tmpDets;
+
+  for (size_t idet = 0; idet < basicComponents().size(); idet++) {
+    double dist2 = (startPos - theDets[idet]->position()).mag2();
+    tmpDets.emplace_back(make_pair(dist2, idet));
```

Comment:
using `make_pair` with `emplace_back` is redundant: just do `emplace_back(dist2, idet)`

Reply:
good remark. Actually I will try to get rid of tmpDets2 as well